src/base/settingsstorage.h

```diff
@@ -62,4 +62,27 @@ public slots:
     mutable QReadWriteLock m_lock;
 };
 
+template<typename T>
+class StorableValue
```

Comment:
@Chocobo1, @glassez:  As a native English speaker, I think `SettingValue` is most clear.